From bd287fa5084f2e153c1044632f9f3d190f090d69 Mon Sep 17 00:00:00 2001
From: Hans Kratz <hans@appfour.com>
Date: Wed, 10 Nov 2021 20:24:33 +0100
Subject: [PATCH] Disable aarch64 outline atomics with musl for now.

The introduced dependency on `getauxval`causes linking
problems with musl, see #89626.

Signed-off-by: Alex Kiernan <alexk@zuma.ai>
Upstream-Status: Backport
---
 .../rustc_target/src/spec/aarch64_unknown_linux_musl.rs     | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/compiler/rustc_target/src/spec/aarch64_unknown_linux_musl.rs b/compiler/rustc_target/src/spec/aarch64_unknown_linux_musl.rs
index 0770f3496c27..6a16b4ce4196 100644
--- a/compiler/rustc_target/src/spec/aarch64_unknown_linux_musl.rs
+++ b/compiler/rustc_target/src/spec/aarch64_unknown_linux_musl.rs
@@ -9,10 +9,6 @@ pub fn target() -> Target {
         pointer_width: 64,
         data_layout: "e-m:e-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128".to_string(),
         arch: "aarch64".to_string(),
-        options: TargetOptions {
-            features: "+outline-atomics".to_string(),
-            mcount: "\u{1}_mcount".to_string(),
-            ..base
-        },
+        options: TargetOptions { mcount: "\u{1}_mcount".to_string(), ..base },
     }
 }
-- 
2.17.1

